### ISSUE TITLE
Functions cannot be called from within a JS script block

### ISSUE THREAD

**Issue by Alvalol** [Scirra/Construct-3-bugs#3308]
## Problem description

Functions cannot be found when using runtime.callFunction()

## Steps to reproduce

1. Create a function (either the new or old implementation)
2. Call it using runtime.callFunction, c3_callFunction or c2_callFunction, depending on context

## Observed result

Function cannot be found

## Expected result

Function is found

## More details

This occurs with all forms of calling functions within a js script block

**Affected browsers/platforms:** Chrome, firefox

**First affected release:**  r164.2 Stable

## System details

Platform information
Browser: Chrome
Browser version: 76.0.3809.132
Browser engine: Chromium
Browser architecture: (unknown)
Context: browser
Operating system: Mac OS X
Operating system version: 10.14_6
Operating system architecture: (unknown)
Device type: desktop
Device pixel ratio: 2
Logical CPU cores: 12
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.132 Safari/537.36
C3 release: r164.2 (stable)
Language setting: en-US

Local storage
Storage quota (approx): 13 gb
Storage usage (approx): 941 mb (7.3%)
Persistant storage: Yes

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Supports highp precision: yes
Vendor: ATI Technologies Inc.
Renderer: AMD Radeon Pro 555X OpenGL Engine
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 2048
Extensions:

EXT_color_buffer_float
EXT_disjoint_timer_query_webgl2
EXT_float_blend
EXT_texture_filter_anisotropic
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
Audio information
System sample rate: 44100 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

Ogg Opus (audio/ogg; codecs=opus)
WebM Opus (audio/webm; codecs=opus)
Ogg Vorbis (audio/ogg; codecs=vorbis)
WebM Vorbis (audio/webm; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

**Comment by AshleyScirra** [comment]
It's not that it can't find it, it's crashing (see the browser console). It works in a script action though, so this only applies to script blocks and you can work around it with a script action. Also note c2_callFunction/c3_callFunction are not supported in the scripting feature, only runtime.callFunction should be used.

**Comment by Alvalol** [comment]
Yes using the script action is a workaround at the moment but script blocks always fail, even with runtime.callFunction.
